src/routes/user.js

```diff
@@ -1,15 +1,12 @@
 import { Router } from 'express'
 import { create, getById, getAll, updateById } from '../controllers/user.js'
-import {
-  validateAuthentication,
-  validateTeacherRole
-} from '../middleware/auth.js'
+import { validateAuthentication } from '../middleware/auth.js'
 
 const router = Router()
 
 router.post('/', create)
 router.get('/', validateAuthentication, getAll)
 router.get('/:id', validateAuthentication, getById)
-router.patch('/:id', validateAuthentication, validateTeacherRole, updateById)
+router.patch('/:id', validateAuthentication, updateById)
```

Comment:
Instead of checking the user's role in the validateCanPatch function, I think we should leave the `validateTeacherRole` middleware in place here

Reply:
We won't be able to use the validateTeacherRole function in the middleware, as this will prevent users being able to update their own profiles this is why we removed it. This ensures that only a teacher or the logged in user will be able to update their user account.

Reply:
Ah, gotcha